src/EdFi.AnalyticsMiddleTier.DataStandard2/Base/MSSQL/0030-View-SectionDim-for-Student-Engagement.sql

```diff
@@ -0,0 +1,57 @@
+﻿-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+IF EXISTS (SELECT 1 FROM INFORMATION_SCHEMA.VIEWS WHERE TABLE_SCHEMA = 'analytics' AND TABLE_NAME = 'SectionDim')
+BEGIN
+    DROP VIEW analytics.SectionDim
+END
+GO
+
+CREATE VIEW [analytics].[SectionDim]
+AS
+     SELECT 
+          CAST(s.SchoolId AS VARCHAR) AS SchoolKey,
+          CONCAT(CAST(s.SchoolId AS NVARCHAR),'-',[s].[ClassPeriodName],'-',[s].[ClassroomIdentificationCode],'-',s.[LocalCourseCode],'-',CAST([s].[TermDescriptorId] AS NVARCHAR),'-',CAST(s.[SchoolYear] AS NVARCHAR),'-',s.[UniqueSectionCode],'-',CAST(s.[SequenceOfCourse] AS NVARCHAR)) AS [SectionKey],
+          CONCAT([Descriptor].[Description],'(',s.[LocalCourseCode],')','-',[Course].[CourseTitle],'(',s.[ClassPeriodName],')',TermType.Description) as Description,
+		  CONCAT([s].[LocalCourseCode],'-',COALESCE([Course].[CourseTitle], '')) AS [SectionName],
+		  [Course].[CourseTitle] AS [SessionName],
+		  [CourseOffering].[LocalCourseCode],
+		  s.SchoolYear, 
+		  EducationalEnvironmentType.Description as EducationalEnvironmentDescriptor,
+		  sch.LocalEducationAgencyId as LocalEducationAgencyKey,
+		  s.LastModifiedDate
+          ,course.CourseTitle
+          ,CONCAT(
+            s.SchoolId
+            ,'-'
+            ,s.SchoolYear
+            ,'-'
+            ,s.TermDescriptorId
+          ) as SessionKey
+   FROM 
+        [edfi].[Section] s
+    INNER JOIN [edfi].[CourseOffering]
+          ON [CourseOffering].[SchoolId] = s.[SchoolId]
+             AND [CourseOffering].[LocalCourseCode] = s.[LocalCourseCode]
+    AND
+        CourseOffering.TermDescriptorId = s.TermDescriptorId
```

Comment:
New join criteria. In Data Standard 2.2, note that `TermDescriptorId` is part of the natural / primary key for `CourseOffering` - and, it is present in `Section`. In Data Standard 3 this was replaced with `SessionName`.

![image](https://user-images.githubusercontent.com/9324390/123850673-ea96f380-d8df-11eb-9282-fbc467e2e401.png)